pkg/credentials/credentials.go

```diff
@@ -28,6 +28,15 @@ import (
 	"k8c.io/kubeone/pkg/apis/kubeone"
 )
 
+// CredentialsType is a type of credentials that should be fetched
+type CredentialsType string //nolint:revive
```

Comment:
Removing this nolint shows:
```
pkg/credentials/credentials.go:32:6: exported: type name will be used as credentials.CredentialsType by other packages, and that stutters; consider calling this Type (revive)
```

Sounds reasonalbe. The name of the package is also a part of the type identity, so why neglect it?

Reply:
@kron4eg Because just `Type` sounds very weird to me. Just `Type` doesn't really describe what it refers to. But when I think a bit more about it, it might make sense. Let me rename it.